### ISSUE TITLE
Unable to run application

### ISSUE THREAD

**Issue by Loksly** [VoltDB/voltdb-client-nodejs#14]
Hi everyone.

I'm trying to deploy both server and client and it seems they are using deprecated api that is no longer working.

I've downloaded lastest voltdb version from github.
Then compiled using JDK 1.8 using these commands: 

```
cd ~
git clone https://github.com/VoltDB/voltdb.git
cd voltdb
ant dist
```

Then temporaly disable transparent huge pages using

```
sudo bash
echo never > /sys/kernel/mm/transparent_hugepage/enabled
echo never > /sys/kernel/mm/transparent_hugepage/defrag
```

I have tested voltdb command as shown in the examples of the home site

```
~/voltdb/bin/voltdb create &
sqlcmd
```

and it seemed to be working.

Then I've deployed this repository

```
cd ~
git clone https://github.com/VoltDB/voltdb-client-nodejs.git
cd voltdb-client-nodejs
npm install
cd examples/voter/voter/
npm install
cd ~/voltdb-client-nodejs/tools/testdb/
export VOLTDB_HOME=~/voltdb
```

but when I run the server:

```
cd ~/voltdb-client-nodejs/tools/testdb/
./run.sh server
```

Next error messages are shown:

```
Warning: DEPRECATED - For future development please use "voltdb compile" (with -h for help).
WARN: Compiling from a project file is deprecated and will be removed in a future release.
------------------------------------------
Successfully created project.xml
Includes schema: ddl.sql

[SP][WRITE] InitTestType
  insert into typetest values ( ?, ?, ?, ?, ?, ?, ?, ?, ?, ? );

[SP][WRITE] Insert
  insert into typetest values ( ?, ?, ?, ?, ?, ?, ?, ?, ?, ? );

------------------------------------------

Catalog contains 0 built-in CRUD procedures.
    Simple insert, update, delete, upsert and select procedures are created
    automatically for convenience.

[SP] Catalog contains 2 single partition procedures.
    Single partition procedures run in parallel and scale
    as partitions are added to a cluster.

------------------------------------------

Full catalog report can be found at file:///home/loksly/voltdb-client-nodejs/tools/testdb/catalog-report.html.

------------------------------------------

ERROR: Extra arguments were provided:
ERROR:   typetest.jar
ERROR:   deployment
ERROR:   deployment.xml
ERROR:   license
ERROR:   /home/loksly/voltdb/voltdb/license.xml
ERROR:   leader
ERROR:   localhost

Usage: voltdb create [ OPTIONS ... ] [ CATALOG ]
```

```
./run.sh srccompile #works flawlessly
```

what is no longer working is:

```
./run.sh catalog
```

I've tried to change it's code to:

```
function catalog() {
    srccompile
   $VOLTDB_HOME/bin/voltdb compile --classpath obj -o $APPNAME.jar -p project.xml
    # stop if compilation fails
    if [ $? != 0 ]; then exit; fi
}

function server(){
    $VOLTDB create $APPNAME.jar -d deployment.xml
    #instead of the next one that uses licence and parameters no longer working
    #$VOLTDB create catalog $APPNAME.jar deployment deployment.xml license $LICENSE leader $LEADER
}

```

After that I run:

```
./run server &
cd ~/voltdb-client-nodejs/examples/voter/voter/
node app.js 
```

And it failed with this stacktrace:

```
TypeError: Cannot read property '0' of undefined
    at VoltMessageManager.initVoter (/home/loksly/voltdb-client-nodejs/examples/voter/voter/models/volt.js:129:31)
    at VoltMessageManager.emit (events.js:118:17)
    at VoltConnection.onRead (/home/loksly/voltdb-client-nodejs/lib/connection.js:202:13)
    at Socket.emit (events.js:107:17)
    at readableAddChunk (_stream_readable.js:163:16)
    at Socket.Readable.push (_stream_readable.js:126:10)
    at TCP.onread (net.js:538:20)
```

when I debugged it shows these warnings:

```
WARN: Procedure Vote was not found. This message is rate limited to once every 60 seconds.
WARN: Procedure Initialize was not found. This message is rate limited to once every 60 seconds.
```

I guess there are a lot of things I'm doing wrong. Can you please guide my next steps?
Thanks,

**Comment by benjaminballard** [comment]
Hi Loksly,

If you are going to compile VoltDB from source, you currently must use JDK 7, not 8.  JDK 8 is supported for running a pre-built VoltDB kit, not for compiling the source.  See the build instructions on the [VoltDB/voltdb repo wiki](https://github.com/VoltDB/voltdb/wiki/Building-VoltDB).

It looks like the tools/testdb application is out of date and needs some script updates to work with the latest version of VoltDB.  I'll be making corrections so that it uses our latest APIs.

The last problem you reported is just that the node.js Voter example client doesn't work if the schema and procedures aren't running in the database.  That should work if you first start the Voter example database following the [README](https://github.com/VoltDB/voltdb/tree/master/examples/voter) instructions under examples/voter in the kit, only instead of running ./run.sh client, you would run:

    cd ~/voltdb-client-nodejs/examples/voter/voter/
    node app.js 

Thanks,
Ben

**Comment by DodiChen** [comment]
Hi Loksly,

Did you resolve your problem? I met the same question, and have no idea to deal with it.

Thanks.
Dodi

**Comment by DodiChen** [comment]
@benjaminballard
sorry, i can't understand what you means about the latest problem. Could you kindly explain it. Thanks you vary much.

**Comment by tianjiaolaozu** [comment]
same issue here:
28 Sep 07:26:48 - Node connected to VoltDB
/root/voltdb-client-nodejs/examples/voter/voter/models/volt.js:125
    var val = results.table[0][0];
                              ^

TypeError: Cannot read property '0' of undefined
    at VoltMessageManager.initVoter (/root/voltdb-client-nodejs/examples/voter/voter/models/volt.js:125:31)
    at emitThree (events.js:97:13)
    at VoltMessageManager.emit (events.js:175:7)
    at VoltConnection.onRead (/root/voltdb-client-nodejs/lib/connection.js:202:13)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:153:18)
    at Socket.Readable.push (_stream_readable.js:111:10)
    at TCP.onread (net.js:536:20)

**Comment by DodiChen** [comment]
I already resolved this issue.
We need to run the run.sh file under voltdb folder rather than voltdb-client-nodejs/tools/testdb/, before run ./run.sh server, please run ./run.sh jars and ./run.sh init first.
After these steps. run node app.js,
every thing will be OK.

**Comment by tianjiaolaozu** [comment]
@DodiChen  awesome

**Comment by Loksly** [comment]
This happened too long ago. I guess it has been solved by now. Whenever I have time to test it again if it doesn't work I will open a new ticket. Thanks for your help.
